src/wpas/wpa_controller.c

```diff
@@ -746,20 +746,25 @@ wpa_controller_destroy(struct wpa_controller **ctrl)
  *
  * @param ctrl The control interface to send the command on.
  * @param name The name of the command to send.
- * @param reply The buffer to hold the reply.
- * @param reply_length The size of the reply buffer. Will be updated with the actual reply length on success.
+ * @param reply The buffer to hold the reply. This must have enough space for a null terminator.
+ * @param reply_length The size of the buffer (including the space for null byte). Will be updated with the actual reply length (minus the null byte) on success.
  * @param fmt The string format of the following arguments constituting the command payload.
  * @param ... The arguments constituting the command payload.
  * @return int 0 if the command was successfully sent. In this case, 'reply'
- * shall contain a *reply_length response from the control interface. -ENOTCON
+ * shall contain a NULL-terminated *reply_length response from the control interface. -ENOTCON
  * is returned if there is no established connection to the control socket.
  * Otherwise, a non-zero value is returned.
  */
 static int
 __wpa_controller_send_commandf(struct wpa_controller *ctrl, const char *name, char *reply, size_t *reply_length, const char *fmt, ...)
 {
+    if (*reply_length == 0) {
```

Comment:
Will want to make sure `reply_length` is non-NULL before dereferencing it. Might as well just bail out of the function if it's NULL since it's needed critically.

Reply:
Got it. I added the change so I will now merge this PR.